### PR TITLE
Incorrect currency displayed on Portfolio Screen

### PR DESCRIPTION
**Pull request by oliver-was-here**
https://www.pivotaltracker.com/story/show/157233356

### CODE REVIEW

File: TradeItIosTicketSDK2/NumberFormatter.swift [hunk 1]
```diff
@@ -2,22 +2,27 @@ class NumberFormatter: NSObject {
...
-            currencyFormatter.negativePrefix = nil
-        }
         currencyFormatter.minimumFractionDigits = minimumFractionDigits
         currencyFormatter.maximumFractionDigits = maximumFractionDigits
-        return currencyFormatter.string(from: number) ?? TradeItPresenter.MISSING_DATA_PLACEHOLDER
+        var formattedCurrency = currencyFormatter.string(from: number)
+        if displayVariance,
+            number.doubleValue > 0.0,
+            let unwrappedFormattedCurrency = formattedCurrency {
+            formattedCurrency = "+\(unwrappedFormattedCurrency)"
```
Comment (oliver-was-here) [inline TradeItIosTicketSDK2/NumberFormatter.swift]: `NumberFormatter#string()` auto prepends `-` out of the box. Presumably we _always_ want to display the `-` sign for negative numbers -- so we need only account for `+`. 

`NumberFormatter` provides some config options (`negativePrefix`, `negativeFormat`, ...), however, they all come with quirks (not respect locale formatting for currencies, and whatever else Apple shoves into their black box.)

File: TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift [hunk 2]
```diff
@@ -44,7 +44,7 @@ class TradeItPortfolioBalanceEquityPresenter {
         var dayReturnString = ""
 
         if let dayAbsoluteReturn = self.balance?.dayAbsoluteReturn {
-            dayReturnString = NumberFormatter.formatCurrency(dayAbsoluteReturn, displayVariance: true, currencyCode: balance?.accountBaseCurrency)
+            dayReturnString = NumberFormatter.formatCurrency(dayAbsoluteReturn, displayVariance: true)
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift]: Why remove this? If the balance returned is not USD then this won't work.

File: TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift [hunk 3]
```diff
@@ -44,7 +44,7 @@ class TradeItPortfolioBalanceEquityPresenter {
         var dayReturnString = ""
 
         if let dayAbsoluteReturn = self.balance?.dayAbsoluteReturn {
-            dayReturnString = NumberFormatter.formatCurrency(dayAbsoluteReturn, displayVariance: true, currencyCode: balance?.accountBaseCurrency)
```
Comment (oliver-was-here) [inline TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift]: If `currencyCode` is `nil`, `currencySymbol` defaults to the symbol dictated by `locale`

Comment (jsom) [reply TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift]: I don't think we want to default to `locale` though - if I'm set to the Thai locale and my balance returns in AUD then I want to see AUD

File: TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift (hunk 2 above)
Comment (oliver-was-here) [reply TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift]: ah shizen, good catch. had mean to circle back with @falnatsheh and think of the best way to handle this.

Options that first come to mind are:
1. `dayReturnString = NumberFormatter.formatCurrency(dayAbsoluteReturn, displayVariance: true, currencyCode: balance?.accountBaseCurrency ?? TradeItPresenter.DEFAULT_CURRENCY_CODE` 
1. Swap out the following code with `currencyFormatter.currencyCode = currencyCode ?? TradeItPresenter.DEFAULT_CURRENCY_CODE`
https://github.com/tradingticket/TradeItIosTicketSDK2/blob/be032b9ebaf53871795758433ba8a032d7cafe28/TradeItIosTicketSDK2/NumberFormatter.swift#L13

I think I'd lean toward option 2.

File: TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift (hunk 3 above)
Comment (oliver-was-here) [reply TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift]: Whoops, this comment was intended for the dif on `TradeItPortfolioAccountsTableViewManager.swift` `L110`

File: TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift (hunk 2 above)
Comment (jsom) [reply TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift]: Maybe the right thing to do is to ignore the value if we can't determine a currency code? Just return `nil`.

Comment (oliver-was-here) [reply TradeItIosTicketSDK2/TradeItPortfolioBalanceEquityPresenter.swift]: good point -- take a look at https://github.com/tradingticket/TradeItIosTicketSDK2/pull/351/commits/a2434727fbda466a1386e9362185a8fcadbac142 and lmk if I'm Swift-ing right

**Comment by oliver-was-here** [comment]
@jsom this is testable, instructions to test are in the [story comments](https://www.pivotaltracker.com/story/show/157233356/comments/190170308)

**Pull request merged**